src/main/java/com/juniorstart/juniorstart/service/UserProfileService.java

```diff
@@ -0,0 +1,107 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.BadRequestException;
+import com.juniorstart.juniorstart.exception.ResourceNotFoundException;
+import com.juniorstart.juniorstart.model.ListUserRole;
+import com.juniorstart.juniorstart.model.User;
+import com.juniorstart.juniorstart.model.UserProfile;
+import com.juniorstart.juniorstart.payload.interfaces.InterfaceChangeRequest;
+import com.juniorstart.juniorstart.repository.UserDao;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+
+import lombok.extern.slf4j.Slf4j;
+import net.bytebuddy.implementation.bytecode.Throw;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.text.WordUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+@Service
+@Slf4j
+public class UserProfileService {
+
+    @Autowired
+    UserProfileRepository userProfileRepository;
+
+    @Autowired
+    UserDao userRepository;
+
+  //  Long userId = new getLogedUserId().userID;
+
+
+    UserProfileService(UserProfileRepository userProfileRepository){
+        this.userProfileRepository=userProfileRepository;
+    }
+
+    //TODO GONNA WORK ON ADD USER PROFILE ON NEXT TASK
+    public UserProfile addUserProfile(UserProfile userProfile){
+
+        //Long userId = new getLogedUserId().userID;
+        //Long tmp = userId;
+        Long userId= 5L;
+
+        Optional<User> foundUser = userRepository.findByPublicId(userId);
+
+        if(foundUser.isPresent()){
+            userProfile.setUser(foundUser.get());
+            userProfile.setPrivateId(foundUser.get().getPrivateId());
+            userProfileRepository.save(userProfile);
+        }
+
+        return userProfile;
+    }
+
+    //TODO GONNA WORK ON UPDATE USER PROFILE ON NEXT TASK
+    public UserProfile updateUserProfile(UserProfile userProfile){
+
+        Long userId= 5L;
+
+
+        Optional<UserProfile> foundUser = userProfileRepository.findById(userId);
+
+        if(foundUser.isPresent()){
+            userProfileRepository.save(userProfile);
+        }
+        return userProfile;
+    }
```

Comment:
To samo co wyżej

Reply:
czemu ma służyć Long userId?

Reply:
@hosu794  funkcjonalność związana z innym taskiem.

Reply:
To usuń z commita poprostu, bo robi się bałagan.